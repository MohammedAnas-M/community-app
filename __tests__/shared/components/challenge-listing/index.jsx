import React from 'react';
import Renderer from 'react-test-renderer/shallow';

const ChallengeListing = require('components/challenge-listing').default;

const selectBucket = jest.fn();
const setFilterState = jest.fn();
const setSort = jest.fn();

const mockData1 = {
  activeBucket: 'abc',
  challenges: [],
  filterState: {},
  lastUpdateOfActiveChallenges: 1500124537142,
  loadingChallenges: false,
  loadingDraftChallenges: false,
  loadingPastChallenges: false,
  selectBucket,
  setFilterState,
  setSort,
  sorts: {},
  auth: {},
};

describe('Matches shallow shapshot 1', () => {
  test('shapshot 1', () => {
    const renderer = new Renderer();

    renderer.render((
      <ChallengeListing {...mockData1} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
