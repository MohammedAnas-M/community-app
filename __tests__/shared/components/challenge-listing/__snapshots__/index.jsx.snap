// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1 shapshot 1 1`] = `
<div
  className="src-shared-components-challenge-listing-___style__ChallengeFiltersExample___3IjeI"
  id="challengeFilterContainer"
>
  <h1
    className="src-shared-components-challenge-listing-___style__tc-title____Rpik"
  >
    CHALLENGES
  </h1>
  <hr
    className="src-shared-components-challenge-listing-___style__tc-seperator___2AnJv"
  />
  <ChallengeTab
    activeBucket="abc"
    previousBucketOfActiveTab={null}
    previousBucketOfPastChallengesTab={null}
    selectBucket={[MockFunction]}
    setPreviousBucketOfActiveTab={[Function]}
    setPreviousBucketOfPastChallengesTab={[Function]}
  />
  <div
    className="src-shared-components-challenge-listing-___style__tc-content-wrapper___1MqlF"
  >
    <div
      className="src-shared-components-challenge-listing-___style__sidebar-container-mobile___3e65d"
    >
      <Connect(SidebarContainer)
        expanding={false}
        recommendedToggle={false}
        setFilterState={[MockFunction]}
        setRecommendedToggle={[Function]}
        setTcoToggle={[Function]}
        tcoToggle={false}
      />
      <Connect(Container)
        communityName={null}
        hidden={true}
        isAuth={false}
        recommendedToggle={false}
        setFilterState={[MockFunction]}
        setRecommendedToggle={[Function]}
        setTcoToggle={[Function]}
        tcoToggle={false}
      />
    </div>
    <Connect(Listing)
      activeBucket="abc"
      auth={Object {}}
      challenges={Array []}
      communityName={null}
      expandTag={null}
      expandedTags={Array []}
      expanding={false}
      filterState={Object {}}
      loadMoreActive={null}
      loadMoreAll={null}
      loadMoreMy={null}
      loadMoreMyPast={null}
      loadMoreOnGoing={null}
      loadMoreOpenForRegistration={null}
      loadMorePast={null}
      loadMoreReviewOpportunities={null}
      loadingPastChallenges={false}
      newChallengeDetails={false}
      openChallengesInNewTabs={false}
      preListingMsg={null}
      prizeMode="money-usd"
      reviewOpportunities={Array []}
      selectBucket={[MockFunction]}
      setFilterState={[MockFunction]}
      setSort={[MockFunction]}
      sorts={Object {}}
    />
  </div>
</div>
`;
